repose-aggregator/components/filters/header-translation-filter/src/main/scala/org/openrepose/filters/headertranslation/HeaderTranslationFilter.scala

```diff
@@ -74,10 +74,19 @@ class HeaderTranslationFilter @Inject()(configurationService: ConfigurationServi
         if (originalHeaderValues.nonEmpty) {
           sourceHeader.getNewName foreach { newHeaderName =>
             originalHeaderValues foreach { originalHeaderValue =>
-              if (quality.isEmpty) {
-                httpRequest.addHeader(newHeaderName, originalHeaderValue)
+              def writeHeaders(writeWithoutQuality: (String, String) => Unit,
+                               writeWithQuality: (String, String, Double) => Unit): Unit = {
+                if (quality.isEmpty) {
+                  writeWithoutQuality(newHeaderName, originalHeaderValue)
+                } else {
+                  writeWithQuality(newHeaderName, withoutQuality(originalHeaderValue), quality.get)
+                }
+              }
```

Comment:
I kind of wonder if it's worth breaking this out into a function. It's usage is pretty narrowly scoped, and whether we do it this way or the "Java" way, I think it comes down to the same thing -- we branch into four distinct paths. That is, the cyclomatic complexity of this method is the same either way, regardless of whether we nest if/then statements or create a 4-way if/else-if statement, where here you've done the former.

I guess I'm wondering what this approach buys us.

Just to be clear, I consider the "Java" way to be:
```
if (quality.isEmpty && sourceHeader.isOverwriteTarget) {
  httpRequest.replaceHeader(newHeaderName, originalHeaderValue)
} else if (quality.isEmpty && !sourceHeader.isOverwriteTarget) {
  httpRequest.addHeader(newHeaderName, originalHeaderValue)
} else if (!quality.isEmpty && sourceHeader.isOverwriteTarget) {
  httpRequest.replaceHeader(newHeaderName, originalHeaderValue, quality.get)
} else if (!quality.isEmpty && !sourceHeader.isOverwriteTarget) {
  httpRequest.addHeader(newHeaderName, originalHeaderValue, quality.get)
}
```